sdk/cognitiveservices/cognitiveservices-qnamaker/src/qnAMakerClientContext.ts

```diff
@@ -24,28 +24,28 @@ export class QnAMakerClientContext extends msRest.ServiceClient {
    * @param credentials Subscription credentials which uniquely identify client subscription.
    * @param [options] The parameter options
    */
-  constructor(endpoint: string, credentials: msRest.ServiceClientCredentials, options?: msRest.ServiceClientOptions) {
-    if (endpoint === null || endpoint === undefined) {
-      throw new Error('\'endpoint\' cannot be null.');
+  constructor(credentials: msRest.ServiceClientCredentials, endpoint: string, options?: msRest.ServiceClientOptions) {
+    if (endpoint == undefined) {
+      throw new Error("'endpoint' cannot be null.");
     }
-    if (credentials === null || credentials === undefined) {
-      throw new Error('\'credentials\' cannot be null.');
+    if (credentials == undefined) {
+      throw new Error("'credentials' cannot be null.");
     }
 
     if (!options) {
       options = {};
     }
-    if(!options.userAgent) {
+
+    if (!options.userAgent) {
       const defaultUserAgent = msRest.getDefaultUserAgentValue();
       options.userAgent = `${packageName}/${packageVersion} ${defaultUserAgent}`;
     }
 
     super(credentials, options);
 
-    this.baseUri = "{Endpoint}";
+    this.baseUri = "{Endpoint}/qnamaker/v4.0";
```

Comment:
the parameterized host extension is honored correctly with the generic autorest.typescript flavor.